### PR TITLE
Add basic pub sub tests

### PR DESCRIPTION
**Pull request by dilanSachi**
## Purpose
$subject
Part of https://github.com/ballerina-platform/ballerina-standard-library/issues/5096

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: native/src/main/java/io/ballerina/lib/ibm.ibmmq/QueueManager.java [hunk 1]
```diff
@@ -111,4 +111,15 @@ public static Object accessTopic(BObject queueManagerObject, BString topicName,
...
+
+    public static Object disconnect(BObject queueManagerObject) {
+        MQQueueManager queueManager = (MQQueueManager) queueManagerObject.getNativeData(NATIVE_QUEUE_MANAGER);
+        try {
+            queueManager.disconnect();
+            return null;
+        } catch (MQException e) {
+            return createError(IBMMQ_ERROR,
+                    String.format("Error occurred while disconnecting queue manager: %s", e.getMessage()), e);
+        }
```
Comment (ayeshLK) [inline native/src/main/java/io/ballerina/lib/ibm.ibmmq/QueueManager.java:123]:
```suggestion
    public static Object disconnect(BObject queueManagerObject) {
        MQQueueManager queueManager = (MQQueueManager) queueManagerObject.getNativeData(NATIVE_QUEUE_MANAGER);
        try {
            queueManager.disconnect();
        } catch (MQException e) {
            return createError(IBMMQ_ERROR,
                    String.format("Error occurred while disconnecting queue manager: %s", e.getMessage()), e);
        }
        return null;
```

File: ballerina/tests/pub_sub_tests.bal [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/test;
+
+@test:Config {}
+function basicPublisherSubscriberTest() returns error? {
+    QueueManager queueManager = check new QueueManager(name = "QM1", host = "localhost", channel = "DEV.APP.SVRCONN");
+    Topic subTopic = check queueManager.accessTopic("dev", "DEV.BASE.TOPIC", OPEN_AS_SUBSCRIPTION, MQSO_CREATE);
+    Topic pubTopic = check queueManager.accessTopic("dev", "DEV.BASE.TOPIC", OPEN_AS_PUBLICATION, MQOO_OUTPUT);
```
Comment (ayeshLK) [inline ballerina/tests/pub_sub_tests.bal:23]:
```suggestion
    Topic subscriber = check queueManager.accessTopic("dev", "DEV.BASE.TOPIC", OPEN_AS_SUBSCRIPTION, MQSO_CREATE);
    Topic publisher = check queueManager.accessTopic("dev", "DEV.BASE.TOPIC", OPEN_AS_PUBLICATION, MQOO_OUTPUT);
```

File: ballerina/tests/pub_sub_tests.bal [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    check pubTopic->put({
+        payload: "Hello World".toBytes()
+    });
+    Message message = check subTopic->get();
+    test:assertEquals(string:fromBytes(message.payload), "Hello World");
+    check subTopic->close();
```
Comment (ayeshLK) [inline ballerina/tests/pub_sub_tests.bal:29]:
```suggestion
    check publisher->close();
    check subscriber->close();
```
check for other places too.

Comment (dilanSachi) [reply ballerina/tests/pub_sub_tests.bal]: Fixed

File: ballerina/tests/pub_sub_tests.bal (hunk 2 above)
Comment (dilanSachi) [reply ballerina/tests/pub_sub_tests.bal]: Fixed

File: native/src/main/java/io/ballerina/lib/ibm.ibmmq/QueueManager.java (hunk 1 above)
Comment (dilanSachi) [reply native/src/main/java/io/ballerina/lib/ibm.ibmmq/QueueManager.java]: Fixed

File: ballerina/queue_manager.bal [hunk 4]
```diff
@@ -22,7 +22,7 @@ public isolated class QueueManager {
         check self.externInit(configurations);
     }
 
-    isolated function externInit(QueueManagerConfiguration configurations) returns Error? = @java:Method {
+    private isolated function externInit(QueueManagerConfiguration configurations) returns Error? = @java:Method {
```
Comment (ayeshLK) [inline ballerina/queue_manager.bal:25]:
```suggestion
    isolated function externInit(QueueManagerConfiguration configurations) returns Error? = @java:Method {
```

Comment (dilanSachi) [reply ballerina/queue_manager.bal:25]: we usually make the extern function private right? Otherwise these are suggested in places like tests

Comment (dilanSachi) [reply ballerina/queue_manager.bal]: Removed

**Review by shafreenAnfar — APPROVED** [review]

**Pull request merged**